pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/LightningKokkosSimulator.cpp

```diff
@@ -480,7 +484,7 @@ auto LightningKokkosSimulator::Measure(QubitIdType wire,
     SetDeviceShots(device_shots);
 
     // It represents the measured result, true for 1, false for 0
-    bool mres = Lightning::simulateDraw(probs, postselect);
+    bool mres = Lightning::simulateDraw(probs, postselect, this->gen);
```

Comment:
Did you only need this for MCM? What about the rest of MPs?

Reply:
I would propose to upgrade `generate_samples` to take a `wires` argument, as in LQubit, and use that via a measurement object. Everything would then use Kokkos' `Random_XorShift64_Pool` RNG. But that is beyond the scope of this PR. If there is a motivation for it, I could quickly put that together however.

Reply:
Absolutely it would be great for the MPs to also use the RNG instance, however it wasn't required to fix our flaky test. Since this requires additional internal device changes, we left it for a follow up.

Reply:
> Absolutely it would be great for the MPs to also use the RNG instance

To leverage Kokkos we need a parallel RNG (and to be straightforwardly performance portable we need Kokkos' RNG). So we could pass the seed to Kokkos' RNG, but not use `this->gen`.

Reply:
> I would propose to upgrade generate_samples to take a wires argument, as in LQubit, and use that via a measurement object. Everything would then use Kokkos' Random_XorShift64_Pool RNG. 

@vincentmr :100:  but, let's accept this PR as is to unblock compilation, and we can get back to this when working on sc-64459.